### PR TITLE
Update use cases in DG

### LINKED ISSUE #251: Add/Update Use Cases in DG
- [ ] Assign by module
 - [ ] Manual
 - [ ] Mark
 - [ ] Unmark
 - [ ] Sort

### PR DESCRIPTION
**Pull request by Benjykoh**
Closes #251 

Use case is already present for assign

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1368,6 +1368,114 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Extensions**
+
+*  1a. The user does not input a specific command
+
+   * 1a1. TAPA provides the list of all available commands
+
+     Use case ends.
+   
+*  1b. The user inputs a command that is not valid (not in the existing list of commands)
```
Comment (brelkh) [inline docs/DeveloperGuide.md:1390]:
```suggestion
*  1b. The user inputs a command that is not valid (not in the existing list of commands).
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1368,6 +1368,114 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. User requests TAPA to provide a description for a particular command, as well as how to use the command.
+2. TAPA provides a short description of what the specified command does.
+3. TAPA provides the format for how to use the command.
+4. TAPA provides an example to show the user how to use the command.
+
+   Use case ends.
+
+**Extensions**
+
+*  1a. The user does not input a specific command
```
Comment (brelkh) [inline docs/DeveloperGuide.md:1384]:
```suggestion
*  1a. The user does not input a specific command.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1368,6 +1368,114 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+   * 1a1. TAPA provides the list of all available commands
```
Comment (brelkh) [inline docs/DeveloperGuide.md:1386]:
```suggestion
   * 1a1. TAPA provides the list of all available commands.
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1368,6 +1368,114 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+  
+   Use case ends.
+
+* 3a. If studentId is given and a student with the given studentId does not exist.
```
Comment (medajet) [inline docs/DeveloperGuide.md:1413]:
```suggestion
* 3a. If student ID is given and a student with the given student ID does not exist.
```

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1368,6 +1368,114 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+   * 3a1. TAPA shows an error message. 
+     
+      Use case ends.
+   
+* 3b. If the index is out of range, that is a non-positive number or greater than the number of tasks assigned to that person.
```
Comment (brelkh) [inline docs/DeveloperGuide.md:1419]:
```suggestion
* 3b. If the index given is out of range, that is, it is a non-positive number, or greater than the number of tasks assigned to that person.
```

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -1368,6 +1368,114 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  
+   * 3b1. TAPA shows an error message. 
+     
+      Use case ends.
+
+* 3c. The specified task with the given index has already been marked as complete.
+
+   * 3b1. TAPA shows an error message.
```
Comment (brelkh) [inline docs/DeveloperGuide.md:1427]:
```suggestion
   * 3c1. TAPA shows an error message.
```

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -1368,6 +1368,114 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. If studentId is given and a student with the given studentId does not exist.
```
Comment (medajet) [inline docs/DeveloperGuide.md:1448]:
```suggestion
* 3a. If student ID is given and a student with the given student ID does not exist.
```

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -1368,6 +1368,114 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+   * 3a1. TAPA shows an error message.
+
+     Use case ends.
+   
+* 3b. If the index is out of range, that is a non-positive number or greater than the number of tasks assigned to that person.
```
Comment (brelkh) [inline docs/DeveloperGuide.md:1454]:
```suggestion
* 3b. If the index is out of range, that is, it is a non-positive number, or greater than the number of tasks assigned to that person.
```

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -1368,6 +1368,114 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+   * 3b1. TAPA shows an error message.
+
+     Use case ends.
+
+* 3c. The specified task with the given index has already been marked as incomplete.
+
+   * 3b1. TAPA shows an error message.
```
Comment (brelkh) [inline docs/DeveloperGuide.md:1462]:
```suggestion
   * 3c1. TAPA shows an error message.
```

**Review by medajet — APPROVED** [review]
LGTM, just changes regarding studentId vs student ID since we are standardising them.

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -1368,6 +1368,114 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+     Use case ends.
+
+**Use case: UC15 - Sorting list of students by their number of incomplete tasks in descending order**
+
+**MSS**
+
+1. User requests to sort students.
+2. TAPA shows a list of students in order sorted by their number of incomplete tasks in descending order.
```
Comment (brelkh) [inline docs/DeveloperGuide.md:1471]:
```suggestion
2. TAPA shows a list of students in order, sorted by their number of incomplete tasks in descending order.
```

**Review by brelkh — APPROVED** [review]
Mostly LGTM! Just some minor typos and errors spotted.

**Pull request merged**